source/exec/docker.d

```diff
@@ -44,31 +52,37 @@ class Docker: IExecProvider
 		logInfo("Maximum Queue Size: %d", maximumQueueSize_);
 		logInfo("Memory Limit: %d MB", memoryLimitMB_);
 		logInfo("Output size limit: %d B", maximumQueueSize_);
-		logInfo("Checking whether Docker is functional and updating Docker image '%s'", DockerImage);
-		logInfo("Using docker binary at '%s'", dockerBinaryPath);
 
-		auto docker = execute([this.dockerBinaryPath_, "ps"]);
-		if (docker.status != 0) {
-			throw new Exception("Docker doesn't seem to be functional. Error: '"
-					~ docker.output ~ "'. RC: " ~ to!string(docker.status));
-		}
+		foreach (dockerImage; DockerImages)
+		{
 
-		auto dockerPull = execute([this.dockerBinaryPath_, "pull", DockerImage]);
-		if (docker.status != 0) {
-			throw new Exception("Failed pulling RDMD Docker image. Error: '" ~ docker.output
-					~ "'. RC: " ~ to!string(docker.status));
-		}
+			logInfo("Checking whether Docker is functional and updating Docker image '%s'", dockerImage);
+			logInfo("Using docker binary at '%s'", dockerBinaryPath);
+
+			auto docker = execute([this.dockerBinaryPath_, "ps"]);
+			if (docker.status != 0) {
```

Comment:
(Also remember to use http://dlang.org/phobos-prerelease/std_exception#basicExceptionCtors)